### DISCUSSION TITLE
I have a new executor - how do I add it to all my `project.json` easily?

### DISCUSSION THREAD

**Discussion by dan-cooke** [nrwl/nx#11383] [General]
This seems like a bit of an issue with `nx` 

If we have custom executors, like `tsc` type-check, we need to manually add it to the `targets` property in every library,

Is there an alternative way to run an executor in every project? 

Or add this to each project one by one.

I could write a generator I suppose - but I am just wondering if i have missed a native way to achieve this?

**Comment by rfprod** [comment]
@dan-cooke here's how I solved it https://github.com/rfprod/nx-ng-starter/tree/main/tools/executors/tsc

There are two executors in the `tsc` namespace:
- check
- configure

A command to configure projects https://github.com/rfprod/nx-ng-starter/blob/main/tools/project.json#L200-L205

The following command will add the executor configuration to all projects
```bash
npx nx run tools:tsc-configure
```

The following command will remove the executor configuration from all projects
```bash
npx nx run tools:tsc-configure --cleanup
```

**Reply by dan-cooke** [reply]
Cool, so custom executor is the way to go. Thanks!

**Reply by Maxim-Mazurok** [reply]
Not sure about that, in my understanding we should be using generators for running local migrations
